crates/runc-shim/src/service.rs

```diff
@@ -81,7 +88,33 @@ impl Shim for Service {
             None => {}
         }
 
-        let address = spawn(opts, &grouping, Vec::new())?;
+        let (child_id, address) = spawn(opts, &grouping, Vec::new())?;
+
+        #[cfg(target_os = "linux")]
```

Comment:
How about split out this as a function?

Reply:
I was planning to do that, but worried about changing too much. Now that you have the same feeling , I'd like to try again :)